pkg/controller/virtualmachine/virtualmachine_controller.go

```diff
@@ -92,50 +92,58 @@ func (r *ReconcilePolicy) Reconcile(request reconcile.Request) (reconcile.Result
 		return reconcile.Result{}, err
 	}
 
+	podFinalizerName := r.poolManager.GetManagerFinalizerName()
 	if instance.ObjectMeta.DeletionTimestamp.IsZero() {
 		log.V(1).Info("The VM is not being deleted",
 			"virtualMachineName", request.Name,
 			"virtualMachineNamespace", request.Namespace)
 		// The object is not being deleted, so if it does not have a finalizer,
 		// then lets add the finalizer and update the object.
-		err = r.addFinalizerAndUpdate(instance, &request)
+		err = r.addFinalizerAndUpdate(instance, &request, podFinalizerName)
 		return reconcile.Result{}, err
 	}
 	// The object is being deleted
 	log.V(1).Info("The VM is being marked for deletion",
 		"virtualMachineName", request.Name,
 		"virtualMachineNamespace", request.Namespace)
-	err = r.removeFinalizerAndReleaseMac(instance, &request)
+	err = r.removeFinalizerAndReleaseMac(instance, &request, podFinalizerName)
 	return reconcile.Result{}, err
 }
 
-func (r *ReconcilePolicy) addFinalizerAndUpdate(virtualMachine *kubevirt.VirtualMachine, request *reconcile.Request) error {
-	if helper.ContainsString(virtualMachine.ObjectMeta.Finalizers, pool_manager.RuntimeObjectFinalizerName) {
+func (r *ReconcilePolicy) addFinalizerAndUpdate(virtualMachine *kubevirt.VirtualMachine, request *reconcile.Request, podFinalizerName string) error {
```

Comment:
Why do you specify that the finalizer name is for "pod"? I think it is irrelevant in this context. This function just adds a finalizer to a VM. It does not matter whether the finalizer is "pod" or "foo".